contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+        } else if (key.startsWith(KAFKA_CONSUMER_PREFIX)) {
+          consumerProps.put(key.replaceAll(KAFKA_CONSUMER_PREFIX, ""), value);
+        } else {
+          producerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+          consumerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+        }
+      }
+    }
+
+    // Init Kafka producer and consumer
+    producer = new KafkaProducer<>(producerProps);
+    consumer = new KafkaConsumer<>(consumerProps);
+    consumer.subscribe(Collections.singletonList(queueName));
+    logger.info("KafkaaObservableQueue initialized for {}", queueName);
+  }
+
+  @Override
+  public Observable<Message> observe() {
+    OnSubscribe<Message> subscriber = getOnSubscribe();
+    return Observable.create(subscriber);
+  }
+
+  @Override
+  public List<String> ack(List<Message> messages) {
+    return messages.stream().map(Message::getId).collect(Collectors.toList());
+  }
+
+  public void setUnackTimeout(Message message, long unackTimeout) {
+  }
+
+  @Override
+  public void publish(List<Message> messages) {
+    publishMessages(messages);
+  }
+
+  @Override
+  public long size() {
+    return 0;
+  }
+
+  @Override
+  public String getType() {
+    return QUEUE_TYPE;
+  }
+
+  @Override
+  public String getName() {
+    return queueName;
+  }
+
+  @Override
+  public String getURI() {
+    return queueName;
+  }
+
+  private List<Message> receiveMessages() {
+    List<Message> messages = new ArrayList<>();
+    try {
+
+      ConsumerRecords<String, String> records = consumer.poll(longPollTimeout);
+      if (records.count() == 0) {
+        return messages;
+      }
+
+      logger.info("polled messages from kafka {}", records.count());
+      records.forEach(record -> {
+        logger.debug("Consumer Record: key {}, value {}, partition {}, offset {}", record.key(),
+            record.value(), record.partition(), record.offset());
+        Message message =
+            new Message(record.key() + record.partition(), String.valueOf(record.value()), "");
+        messages.add(message);
+      });
+      consumer.commitAsync();
+    } catch (KafkaException e) {
+      logger.error("kafka consumer message polling failed.", e);
+    }
+    return messages;
+  }
+
+  private void publishMessages(List<Message> messages) {
+
+    if (messages == null || messages.isEmpty()) {
+      return;
+    }
+    for (Message message : messages) {
+      final ProducerRecord<String, String> record =
+          new ProducerRecord<>(queueName, message.getId(), message.getPayload());
+
+      RecordMetadata metadata;
+      try {
+        metadata = producer.send(record).get();
+        logger.debug("Producer Record: key {}, value {}, partition {}, offset {}", record.key(),
+            record.value(), metadata.partition(), metadata.offset());
+      } catch (InterruptedException | ExecutionException e) {
+        Thread.currentThread().interrupt();
```

Comment:
do you want to interrupt thread when ExecutionException is thrown?

Reply:
corrected.